File: 209. Minimum Size Subarray Sum/209. Minimum Size Subarray Sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function (target, nums) {
  let minLength = Infinity;
  let currSum = 0;
  let left = 0;

  for (let i = 0; i < nums.length; i++) {
    currSum += nums[i];

    while (currSum >= target) {
      minLength = Math.min(minLength, i - left + 1);
      currSum -= nums[left];
      left++;
    }
  }

  return minLength < Infinity ? minLength : 0;
};

File: 209. Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_size = float("inf")
        left = 0
        summ = 0

        for right in range(len(nums)):
            summ += nums[right]

            while summ >= target:
                min_size = min(min_size, right - left + 1)
                summ -= nums[left]
                left += 1

        return min_size if min_size != float("inf") else 0
